src/managers/config.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Manager for handling configuration building + writing."""
+
+import logging
+
+from ops.model import ConfigData
+
+from core.cluster import ClusterState
+from core.workload import WorkloadBase
+from literals import CONFIG_FILE
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_PROPERTIES = """
+initial-cluster-token: 'etcd-cluster'
+snapshot-count: 10000
+heartbeat-interval: 100
+election-timeout: 1000
+quota-backend-bytes: 0
+max-snapshots: 5
+max-wals: 5
+strict-reconfig-check: false
+enable-pprof: true
+proxy: 'off'
+proxy-failure-wait: 5000
+proxy-refresh-interval: 30000
+proxy-dial-timeout: 1000
+proxy-write-timeout: 5000
+proxy-read-timeout: 0
+force-new-cluster: false
+auto-compaction-mode: periodic
+auto-compaction-retention: "1"
+"""
+
+# these config properties are not used at the moment
+# they are only listed here for completeness
+TLS_PROPERTIES = """
```

Comment:
Similar comment, regarding yaml parsing - we can have this defined as a python dictionary and dump it into the config file using the yaml lib

Reply:
See above.